src/crimson/osd/heartbeat.cc

```diff
@@ -292,7 +292,26 @@ seastar::future<> Heartbeat::handle_ping(crimson::net::ConnectionRef conn,
       service.get_mnow(),
       service.get_up_epoch(),
       min_message);
-  return conn->send(std::move(reply));
+  return conn->send(std::move(reply)
+  ).then([this, m, conn] {
+    return maybe_share_osdmap(conn, m);
+  });
+}
+
+seastar::future<> Heartbeat::maybe_share_osdmap(
+  crimson::net::ConnectionRef conn,
+  Ref<MOSDPing> m)
+{
+  const osd_id_t from = m->get_source().num();
+  auto found = peers.find(from);
+  if (found == peers.end()) {
+    return seastar::now();
+  }
+  auto &peer = found->second;
+  if (m->map_epoch < peer.get_epoch()) {
```

Comment:
Just to let you know, I will be also looking at this PR.
After discussing with Sam, since `m->map_epoch` is already sent from the peer, it appears that the if case should be:
`if (service.get_map()->get_epoch() > m->map_epoch) {`


However, I'm still trying working on understanding how did `m->map_epoch` was earlier than `peer.get_epoch()`.

Reply:
Please let me know if I can push these commits into your PR, I'm working on testing them.
https://github.com/Matan-B/ceph/compare/wip-share-map-over-heartbeat...Matan-B:ceph:wip-share-map-over-heartbeat-edit?expand=1

Reply:
@Matan-B Hi, Sorry for the late reply.

I was occupied recently and haven't got the time to look into this PR. I'm really sorry for that.

Please feel free to push commits here, thanks very much:-)

Reply:
I will schedule a test run before pushing here, Thanks!